### PR TITLE
replaced Session->setFlash with FlashComponent

### PR DESCRIPTION
**Pull request by xhs345**
Flash method description was listed twice

### CODE REVIEW

**Comment by antograssiot** [comment]
maybe we could keep both and add a comment that mention before/after 2.7 ?

**Comment by ravage84** [comment]
The ``SessionComponent::setFlash()`` is deprecated:

http://api.cakephp.org/2.7/class-SessionComponent.html#_setFlash

In the past, we removed stuff that is deprecated from the book and repalced it with the new stuff.

But since the whole shebang has been in the book for quite some time we should add proper notes, or have a central place where we show how to use the new sutff along the old one side by side.

**Comment by xhs345** [comment]
I'm not entirely sure what you mean by adding proper notes. The Flash component is already explained and I don't think it's practical to add a notice to every place where setFlash is replaced by the Flash component (that would result in a lot of those notices). Maybe I'm also not to worries because replacing the method is not really a "breaking" change

**Comment by dereuromark** [comment]
I am ok with only showing the new "correct" way of doing things in this case.

File: en/core-libraries/components/authentication.rst [hunk 1]
```diff
@@ -195,12 +195,7 @@ working with a login form could look like::
                 // Prior to 2.3 use
                 // `return $this->redirect($this->Auth->redirect());`
             }
-            $this->Session->setFlash(
-                __('Username or password is incorrect'),
-                'default',
-                array(),
-                'auth'
-            );
+            $this->Flash->error(__('Username or password is incorrect'));
```
Comment (markstory) [inline en/core-libraries/components/authentication.rst]: I would include the 'old' way to do this here as well. The docs don't only cover 2.7+.

File: en/core-libraries/components/authentication.rst [hunk 2]
```diff
@@ -325,8 +320,8 @@ need to add the following code to your layout. Add the following two
 lines to the ``app/View/Layouts/default.ctp`` file in the body section
 preferable before the content_for_layout line. ::
 
-    echo $this->Session->flash();
-    echo $this->Session->flash('auth');
+    echo $this->Flash->render();
+    echo $this->Flash->render('auth');
```
Comment (markstory) [inline en/core-libraries/components/authentication.rst]: Same here. Include both with a note about the versions. Something like:

```php
// Before CakePHP 2.7
echo $this->Session->flash();

// CakePHP 2.7+
echo $this->Flash->render();
```

**Comment by xhs345** [comment]
Ok, please have a look if this looks better

**Comment by markstory** [comment]
Thanks @xhs345 :+1:

**Pull request merged**